qiskit/circuit/gate.py

```diff
@@ -42,6 +45,26 @@ def to_matrix(self):
         """
         raise QiskitError("to_matrix not defined for this {}".format(type(self)))
 
+    def power(self, exponent):
+        """Creates a unitary gate as `gate^exponent`.
+
+        Args:
+            exponent (float): Gate^exponent
+
+        Returns:
+            UnitaryGate: To which `to_matrix` is self.to_matrix^exponent.
+        """
+        from qiskit.extensions.unitary import UnitaryGate  # pylint: disable=cyclic-import
```

Comment:
I'm a bit concerned by this class hierarchy. The `Gate` class is now aware of its derived class `UnitaryGate`. So if one day you decide to remove UnitaryGate, you break its parent. This is against object oriented principles. But I don't have a better solution now.

Reply:
Should standard gates subclass `UnitaryGate`? The method can be in `UnitaryGate` in that case.

Reply:
let's leave it at this for now, but clearly we need better class hierarchy.